org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java

```diff
@@ -230,6 +231,7 @@ private CompletionList computeContentAssist(ICompilationUnit unit, CompletionPar
 			}
 		}
 
+		CompletionProposalUtils.addStaticImportsAsFavoriteImports(unit);
```

Comment:
Seems like here, it's updating the static import before each completion. Then is it still necessary to update on opening document?

Reply:
When you only open a Java file without editing it, we can still keep its static imports added as favorite imports.

Reply:
I see, then it can benefit other files. Makes sense.